proms-resources/backend/src/main/java/io/uhndata/cards/proms/slacknotifications/SlackNotificationsTask.java

```diff
@@ -0,0 +1,169 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.uhndata.cards.proms.slacknotifications;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.json.Json;
+import javax.json.JsonObject;
+
+import org.apache.sling.api.resource.LoginException;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceResolverFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.uhndata.cards.httprequests.HttpRequests;
+import io.uhndata.cards.performancenotifications.PerformanceUtils;
+
+public class SlackNotificationsTask implements Runnable
+{
+    private static final String SLACK_PERFORMANCE_URL = System.getenv("SLACK_PERFORMANCE_URL");
+
+    /** Default log. */
+    private static final Logger LOGGER = LoggerFactory.getLogger(SlackNotificationsTask.class);
+    private static final String LABEL_TODAY = "today";
+    private static final String LABEL_TOTAL = "total";
+
+    /** Provides access to resources. */
+    private final ResourceResolverFactory resolverFactory;
+
+    SlackNotificationsTask(final ResourceResolverFactory resolverFactory)
+    {
+        this.resolverFactory = resolverFactory;
+    }
+
+    private String buildNotificationLine(String prevValue, Map<String, Long> statMap, String name)
+    {
+        String notificationLine = prevValue;
+        if (notificationLine.length() > 0) {
+            notificationLine += "\n";
+        }
+        notificationLine += "*";
+        notificationLine += name;
+        notificationLine += "*";
+        notificationLine += " -- _Today_: ";
+        notificationLine += statMap.get(LABEL_TODAY);
+        notificationLine += ", _Total_: ";
+        notificationLine += statMap.get(LABEL_TOTAL);
+        return notificationLine;
+    }
+
+    private void postToSlack(String slackUrl, String msg)
+    {
+        try {
+            JsonObject slackApiReq = Json.createObjectBuilder()
+                .add("text", msg)
+                .build();
+            HttpRequests.getPostResponse(slackUrl, slackApiReq.toString(), "application/json");
+        } catch (IOException e) {
+            LOGGER.warn("Failed to send performance update to Slack");
+        }
+    }
+
+    @Override
+    public void run()
+    {
+        LOGGER.warn("Running SlackNotificationsTask");
```

Comment:
This should be  `debug` message.

```suggestion
        LOGGER.debug("Running SlackNotificationsTask");
```